openhands/cli/tui.py

```diff
@@ -700,20 +700,207 @@ def _(event: KeyPressEvent) -> None:
         return '/exit'
 
 
-async def read_confirmation_input(config: OpenHandsConfig) -> str:
+def _generate_single_command_pattern(command: str) -> str:
+    """Generate a regex pattern for a single command that can match similar commands.
+
+    Args:
+        command: The single command to generate a pattern for.
+
+    Returns:
+        str: A regex pattern that matches similar commands.
+    """
+    import re
+
+    # Split command into parts
+    parts = command.split()
+    if not parts:
+        return f'^{re.escape(command)}$'
+
+    # First part is the command itself (keep exact)
+    pattern_parts = [re.escape(parts[0])]
+
+    # Process remaining parts
+    for part in parts[1:]:
+        escaped_part = re.escape(part)
+
+        # Replace file extensions with generic patterns
+        extensions = [
+            'py',
+            'js',
+            'ts',
+            'txt',
+            'md',
+            'json',
+            'yaml',
+            'yml',
+            'toml',
+            'html',
+            'css',
+        ]
+        for ext in extensions:
+            if escaped_part.endswith(f'\\.{ext}'):
+                # Replace filename.ext with any filename with extension
+                escaped_part = '[^\\s]+\\.[a-zA-Z0-9]+'
+                break
+
+        # Handle special cases like .py (starting with dot) - but only if it's a standalone extension
+        if (
+            escaped_part.startswith('\\.')
+            and len(escaped_part) > 2
+            and '/' not in escaped_part
+        ):
+            # Check if this looks like a file extension pattern (e.g., .py, .txt)
+            ext_part = escaped_part[2:]  # Remove the \.
+            if ext_part.isalpha() and len(ext_part) <= 5:  # Common extension length
+                escaped_part = '\\.[a-zA-Z0-9]+'
```

Comment:
@Openhands Let's just accept EVERYTHING after the first command, like `grep [arbitrary text is ok]` and `glob [arbitrary text is ok]`